packages/core/core/src/services/document-service/internationalization.ts

```diff
@@ -36,7 +37,15 @@ const localeToLookup: Transform = (contentType, params) => {
   }
 
   if (params.locale) {
-    return assoc(['lookup', 'locale'], params.locale, params);
+    const isValidLocale = typeof params.locale === 'string';
+
+    if (isValidLocale) {
+      return assoc(['lookup', 'locale'], params.locale, params);
+    }
+
+    throw new errors.ValidationError(
+      `Invalid locale param ${params.locale} provided. Document locales must be strings.`
+    );
   }
 
   return params;
```

Comment:
Primary suggestion is `String(params.locale)` (or maybe there's something safer/better) since we're trying to display as string something we explicitly know isn't a string.

But could we rewrite it more like this to make it less nested?

```
const localeToLookup: Transform = (contentType, params) => {
  if (
    !params.locale ||
    !strapi.plugin('i18n').service('content-types').isLocalizedContentType(contentType)
  ) {
    return params;
  }

  if (typeof params.locale !== 'string') {
    throw new errors.ValidationError(
      `Invalid locale param ${String(params.locale)} provided. Document locales must be strings.`
    );
  }

  return assoc(['lookup', 'locale'], params.locale, params);
};
```

Reply:
And maybe a comment about why `'*'` is allowed here when it's not in the others?